pkg/controller/migration/convert/network.go

```diff
@@ -437,16 +440,213 @@ func getCNIPlugin(c *components) (operatorv1.CNIPluginType, error) {
 	}
 }
 
-func handleIPPool(c *components, install *Installation) error {
+func handleIPPools(c *components, install *Installation) error {
```

Comment:
could you add a comment to this function explaining what you and casey agreed on for the design here? My understanding is it's this:
```
// handleIPPools sets the installation's initialPools field. We first check the CALICO_IPV4POOL_* env vars on calico-node. 
// However, the user may have migrated off of that pool since it is only initially used when no other pool exists.
// As such, we make a client request to the pools in the datastore, and find one ipv4 and one ipv6 pool if at least one of each exists.
// When executing this client request, we prefer the pool if it matches the CALICO_IPV4POOL_* env var. However, we will pick any other if we cannot find one that matches.
```

Reply:
Do we really need logic that complicated? Why not just something like this:

- We query all the IP pools
- If pools with the default-vX-pool name exist, use those.
- Otherwise, pick the first IPv4 and first IPv6 pools that we find.

Reply:
Currently the code does what you've suggested @caseydavenport  with the added step of 1st trying to pick the one specified in the `*POOL_CIDR` variable. So I should drop the POOL_CIDR search?

Reply:
I don't mind much either way - I think if I was writing from scratch I wouldn't bother trying to compare the pools that exist with the ones in the POOL_CIDR variables, but not a problem per-se, just slightly simpler / easier to understand.